### ISSUE TITLE
0.21.0 iOS build failed when Google Maps are not included

### ISSUE THREAD

**Issue by radubatori** [react-community/react-native-maps#2159]
### Is this a bug report?
Yes

### Have you read the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md)?
Yes

### Environment
react: 16.3.0
react-native: 0.54.4
react-native-maps: 0.21.0
iOS only, without Google Maps

### Expected Behavior
The build succeeds.

### Actual Behavior
The iOS build crashes with `AIRGoogleMapCircle.h:7:9: fatal error: 'GoogleMaps/GoogleMaps.h' file not found` even if I do not use Google Maps.
The cause is commit https://github.com/react-community/react-native-maps/commit/d22f96f9115f10b50085206463143c421f8d948d where some Google-depending sources have been added to `AirMaps.xcodeproj/project.pbxproj`.

For now I got things moving again with a npm postinstall script that deletes the references:
`"postinstall": "sed -i '' '/Google.*\\.[h|m]/d' node_modules/react-native-maps/lib/ios/AirMaps.xcodeproj/project.pbxproj"`

### Reproducible Demo
I expect that any iOS project that does not use Google Maps will have the same problem.

**Comment by rborn** [comment]
@radubatori did you remove from podfile all the Google references as per installation instructions?
You might also need to clean your xcode project and run pod install again if you had the google refs on before.

**Comment by radubatori** [comment]
I did not use Google Maps from the star, so the podfile does not reference 'react-native-google-maps' or 'GoogleMaps'.
The same result is obtained if I do not use pods and add AirMaps as a library directly.

**Comment by rborn** [comment]
@rafaellincoln could you please have a look? 🤗

**Comment by TheWrongAlice** [comment]
I just upgraded from 0.20.1 to 0.21.0 and I appear to be having this issue as well.

**Comment by rafaellincoln** [comment]
@radubatori I could not simulate the error, can you help me simulate this error?

I created a repository to perform the tests.
https://github.com/rafaellincoln/rn-maps-ios

**Comment by radubatori** [comment]
Thank you for your support, but meanwhile I had to downgrade to 0.20.1 because of gradle 4 incompatibilities with some of my project's dependencies.

Still, I suspect the issue occurs only when having the AirMaps library linked directly and I somehow missed that during the upgrade. However, even if this installation mode is not officially supported, it still works fine with version 0.20.1 (at least if you do use Google Maps), so it is possible this issue will prevent users that do not wish to use pods from upgrading.

Thank you again for your support, I will close this issue until I attempt to upgrade again and have a clean repository where the issue can be reproduced.

**Comment by TheWrongAlice** [comment]
Should I open an issue of my own? I'm not sure if the bug i'm experiencing is the same or different.

**Comment by radubatori** [comment]
@TheWrongAlice If running `sed -i '' '/Google.*\\.[h|m]/d' node_modules/react-native-maps/lib/ios/AirMaps.xcodeproj/project.pbxproj` (macos) will fix your problem, then it's probably the same issue.

**Comment by meylor** [comment]
I ran into the same issue using:

```
react: ^16.3.0
react-native: ^0.54.2
react-native-maps: ^0.21.0
```

without having the GoogleMaps headers configured for iOS. Fixed by applying the`.patch` file below to `react-native-maps` tag `v0.21.0` during postinstall.

https://gist.github.com/meylor/442cf17ae6ff6213b358866dde4f5b77

The sed command didn't fix the issue for me.

**Comment by TheWrongAlice** [comment]
@meylor I got some type of error running that patch. Now I got 33 new build errors. Did I use it wrong?
```
project_root$ cd node_modules/react-native-maps/
project_root/node_modules/react-native-maps$ patch -p1 < react-native-maps-remove-google-support.patch
patching file ./lib/ios/AirMaps.xcodeproj/project.pbxproj
patch unexpectedly ends in middle of line
Hunk #4 succeeded at 240 with fuzz 1.
```

**Comment by arichnad** [comment]
Can you reopen this ticket?  I'm also having this problem.  The sed line that @radubatori suggests does fix this problem for me. (@radubatori you should remove one of the backslashes in your sed line)

**Comment by radubatori** [comment]
I've reopened the issue, but I cannot provide a clean environment to reproduce it.

@arichnad The extra backslash is there to escape the backslash in `package.json`. If the command is ran directly, then yes, one backslash has to be removed.

**Comment by meylor** [comment]
@TheWrongAlice are you using `react-native-maps: 0.21.0` ?

**Comment by TheWrongAlice** [comment]
@meylor Yes that's right.

**Comment by meylor** [comment]
@TheWrongAlice here are the versions from my `package.json`
```
{
  "name": "MobileApp",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "postinstall": "scripts/postinstall.sh",
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "react": "16.3.0",
    "react-native": "0.54.2",
    "react-native-maps": "0.21.0"
    ...
  },
  "devDependencies": {
    ...
  },
  "jest": {
    "preset": "react-native"
  }
}
```

postinstall.sh is is just doing `patch -p1 < $patchfile` ; perhaps you're not having the same issue?

**Comment by TheWrongAlice** [comment]
@meylor I got your solution to work now (I guess I was doing something wrong the first time).
My build is succeeding! :balloon: 
I hope that this can be fixed in the master branch soon.

**Comment by rborn** [comment]
@radubatori @rafaellincoln @meylor could you guys collaborate and make a PR to solve this issue 🤗 ❤️

**Comment by hlbp** [comment]
@TheWrongAlice  what are you doing wrong? i have the same problem like you.  im getting 

`Helmers-Mac:react-native-maps helmer$ patch -p1 < react-native-maps-remove-google-support.patch
patching file ./lib/ios/AirMaps.xcodeproj/project.pbxproj
patch unexpectedly ends in middle of line
Hunk #4 succeeded at 240 with fuzz 1.
 `

**Comment by JCMais** [comment]
Same issue here, using the patch mentioned by @meylor fixed it for me.

**Comment by meylor** [comment]
@rborn @rafaellincoln I can try to take a look at making a PR this weekend. I don't think that removing the AirGoogleMaps dependencies from `project.pbxproj` is the right approach since someone may actually want to use GoogleMaps. I'm guessing that most people are having an issue because they're not using CocoaPods, which is a requirement in the installation instructions. 

Simple solution should be to add instructions for manually adding the GoogleMaps header files. I haven't yet looked at the source, but it seems odd that the default for not using GoogleMaps is to not have the Pod dependencies, which are require a requirement in the `project.pbxproj` file.

```
  pod 'react-native-maps', path: rn_maps_path
  pod 'react-native-google-maps', path: rn_maps_path  # Remove this line if you don't want to support GoogleMaps on iOS
  pod 'GoogleMaps'  # Remove this line if you don't want to support GoogleMaps on iOS
  pod 'Google-Maps-iOS-Utils' # Remove this line if you don't want to support GoogleMaps on iOS
```

**Comment by JCMais** [comment]
One way to not use GoogleMaps is to use ${inherited} on the header / framework search paths.

If that was included it would be possible to simple add the framework directly and set their path on the settings of the app.

**Comment by douglasjunior** [comment]
Estou usando sem cocoapods, após a atualização para 0.21.0 este problema aparece.

É possível consertar sem cocoapods?

**Comment by kidmysoul** [comment]
Is there any  installation instructions for people who do not use cocoapods?
Cann't get the new version work in ios because of the GoogleMap.h not fount error.

**Comment by hamsddxn** [comment]
remove all google maps reference in build phases -> compile source in Librairies/AirMap/Build Phases

**Comment by atkit** [comment]
@hamsddxn this solution means it needs to be done every time yarn install re-installs libraries and it will not work for CI-s. 

I use cocoa-pods in my projects but I prefer to keep react-native out of it so far, as there are other libraries that depend on react and don't support cocoa-pods.

**Comment by duro** [comment]
I would love to see some installation instructions that do not require CocoaPods. No matter how complicated they may be.

**Comment by kidmysoul** [comment]
I tried install without cocoapod and manually install GoogleMap SDK(because I want to use google map), but it's  really not good, it wasted me a lot of time.

the point is that you have to install [google-maps-ios-utils](https://github.com/googlemaps/google-maps-ios-utils) to use react-native-map(otherwise you will get errors) but that utils doesnot have  manual installation instructions except cocoapod. 

so finally I have to use cocoapod to use this library.

**Comment by duro** [comment]
We managed to get it installed without CocoaPods! 

cc: @theonetheycallneo maybe we could share what we did with this thread to help others looking to do a pure manual install.

**Comment by zzimbler** [comment]
@duro would love to see manual install instructions sans cocoa pods

**Comment by Freddy03h** [comment]
@duro yes please share your experience !

**Comment by douglasjunior** [comment]
@theonetheycallneo, Does this tutorial work with Apple Maps, or just with Google Maps?

**Comment by tavriaforever** [comment]
@theonetheycallneo Hi! 
I tried your instruction, but maybe something missed. 

What I am doing. 

1. Install react-native-map via yarn (it help to install for react-native@0.55.x
2. Download ios sdk zip 2.0.1 from https://developers.google.com/maps/documentation/ios-sdk/start and extract it in my temporarly folder.

**question:** what means the line ```$(SRCROOT)/Frameworks/Google-Maps-iOS-Utils``` in this step?

3. Open xcode, I have had already folder `Frameworks` and it contains `libReact.a`, so I added `GoogleMaps.framework`, `GoogleMapsCore.framework` and `GoogleMapsBase.framework` from unzipped ios sdk folder. 

Drag this three frameworks to `Embedded Binaries`

question: What it means? I don`t have a  folder `Google-Maps-iOS-Utils` in xcode `Frameworks`.

- Search Xcode Build Settings for "Header Paths" and add your new framwork folder from above:
```
$(SRCROOT)/Frameworks/Google-Maps-iOS-Utils
```

![rbtse xcodeproj 2018-05-24 09-42-02](https://user-images.githubusercontent.com/975906/40468690-cb0a8d2c-5f36-11e8-907a-d41300ae0b22.png)

**notice:** I noticed, that after that, there were no changes in `Build Phases/Compile Sources`, it contains only `main.m` and `AppDelegate.m`.

![airgooglemap m 2018-05-24 09-36-42](https://user-images.githubusercontent.com/975906/40468403-ff21f574-5f35-11e8-8af8-e46cea621093.png)

4. Add AirMaps folder from node_modules/react-native-maps/lib/ios. I drag from finder to `Frameworks` folder in xcode. You could see it in left sidebar in above screenshot.
```
$(GITROOT)/node_modules/react-native-maps/lib/ios/AirGoogleMaps
$(GITROOT)/node_modules/react-native-maps/lib/ios/AirMaps
```

**notice:** In Build Phases/Compile Sources` there are a lot of files after step 4.

5. After that I was trying to build an ios app. 

But I got an error:

![airgooglemap m 2018-05-24 09-47-20](https://user-images.githubusercontent.com/975906/40469138-37e3032e-5f38-11e8-844b-ee22c94c1945.png)

I research a problem and found that this is a dependency from https://github.com/googlemaps/google-maps-ios-utils but I don`t have it. How I could install it?

**Comment by mikemorris** [comment]
This issue looks like a duplicate of https://github.com/react-community/react-native-maps/issues/2270.

I opened a PR at https://github.com/react-community/react-native-maps/pull/2310 which coincidentally appears to implement the same patch that @meylor came up with, please test it to see if it fixes this issue for y'all!

@radubatori The gradle 4 issue you mentioned sounds like it could be https://github.com/facebook/react-native/issues/16906

**Comment by billtlee** [comment]
Hi, so is there a work around?  I saw a few mentions of removing AirGoogleMaps from AirMaps.xcodeproj.  What is the correct way to do this?  I assume I have to do it in xcode?  Thanks!

**Comment by dj-hedgehog** [comment]
@billtlee 
@radubatori suggested a workaround himself.

I automated it (for CI etc.) by having this in the `package.json`:

```
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest",
    "postinstall": "cross-os fix-maps"
  },
  "cross-os": {
    "fix-maps": {
      "darwin": "sed -i '' '/Google.*\\.[h|m]/d' node_modules/react-native-maps/lib/ios/AirMaps.xcodeproj/project.pbxproj"
    }
  },
```

and of course `npm install -S cross-os`

It basically removes all references to Google Maps in the faulty `AirMaps.xcodeproj/project.pbxproj`

**Comment by billtlee** [comment]
@dj-hedgehog Thx!

**Comment by pushpendra65** [comment]
@meylor the gist worked for me. 
The project succeeded to build. 
But It gives me error while archiving. Any suggestion?

**Comment by rborn** [comment]
#2396  was merged, please test with the latest git version. 
Please keep in mind that you might have to setup the module again as now it works with `react-native link`
Read the install manual carefully

**Comment by shide1989** [comment]
@dj-hedgehog worked ! thx !

**Comment by nolan12345** [comment]
Add pod 'GoogleMaps', '~> 2.7.0' and then pod install solves the problem

**Comment by dinukasal** [comment]
@nolan12345 it doesn't work

**Comment by JulienCorb** [comment]
nothing has worked for me so far :(
